hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1255,8 +1265,23 @@ public void rename(final Path source, final Path destination,
       /*
        * Fetch the list of blobs in the given sourcePath.
        */
-      List<BlobProperty> srcBlobProperties = getListBlobs(source, null,
-          tracingContext, null, true);
+      String listSrc = source.toUri().getPath() + (source.isRoot()
```

Comment:
Since we are concatenating strings, using string builder would be a better option

Reply:
taken.